File: Pattern42.js

const readline = require('readline');

const FIRST_LETTER = 'A';
const LETTERS_ALPHABET = 26;

const input = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

input.question("Please enter a number\n", (n) => {
  for (let i = 0; i < n * 2; i++) {
    for (let j = 0; j < n - i && i < n; j++) {
      let letter = String.fromCharCode(FIRST_LETTER.charCodeAt(0) + (j % LETTERS_ALPHABET));
      process.stdout.write(`${letter} `);
    }

    for (let j = 0; j < i - n + 1 && i >= n; j++) {
      let letter = String.fromCharCode(FIRST_LETTER.charCodeAt(0) + (j % LETTERS_ALPHABET));
      process.stdout.write(`${letter} `);
    }

    process.stdout.write("\n");
  }

  input.close();
});

File: Pattern42.py
FIRST_LETTER = ord('A')
LETTERS_ALPHABET = 26

n = int(input("Please enter a number\n"))

for i in range(n * 2):
    if i < n:
        for j in range(n - i):
            letter = chr(FIRST_LETTER + (j % LETTERS_ALPHABET))
            print(letter, end=' ')
    else:
        for j in range(i - n + 1):
            letter = chr(FIRST_LETTER + (j % LETTERS_ALPHABET))
            print(letter, end=' ')
    print()
